CRM/Core/Config.php

```diff
@@ -93,12 +93,11 @@ public static function &singleton($loadFromDB = TRUE, $force = FALSE) {
           // initialized before calling applyLocale.
           $sess = \CRM_Core_Session::singleton();
           $sess->initialize();
-          if ($sess->getLoggedInContactID()) {
-            // Apply user's timezone.
-            if (is_callable([self::$_singleton->userSystem, 'setMySQLTimeZone'])) {
-              self::$_singleton->userSystem->setMySQLTimeZone();
-            }
-          }
+          // Apply timezone for this session (will use logged in user's timezone if set)
+          $sessionTime = self::$_singleton->userSystem->getTimeZoneString();
+          date_default_timezone_set($sessionTIme);
```

Comment:
Typo "TIme" => "Time"

But also I think the is_callable might be necessary? If you look up a little bit this is in a block that at first glance is strange to do `if CIVICRM_UF` instead of using the power of OO to call userSystem->something() so that it only happens in standalone, but if I remember right it's like that because all this can happen really early on. The reason wasn't mentioned at https://github.com/civicrm/civicrm-core/pull/29282 but it rings a bell.

Reply:
argh typo must have crept in with some last minute "better name for this variable-ing"...

Reply:
Interesting about the lack of OO. I can't see how it wouldn't be up there -- the userSystem object is initialised in Container::boot which is called on the line above.

I had been thinking "everything inside this if standalone should be moved to a single function on CRM_Utils_System_Standalone" -- but... I think it could be worth a try moving it to a more general hook.

Reply:
Just noticed we also unguardedly call `userSystem->isLoaded` a couple of lines later

Reply:
I like the point you raised about a default timezone in the other PR. 

The easy place to add it right now would be as a Standaloneusers extension setting, though possibly a bit confusing place for it to live.